src/hooks/useGetPurposesActions.ts

```diff
@@ -92,22 +92,38 @@ function useGetPurposesActions(purpose?: DecoratedPurpose) {
     action: handleUpdateDailyCalls,
   }
 
-  const hasVersion = Boolean(purpose.mostRecentVersion)
-
   const availableActions: Record<PurposeState, Array<ActionItem>> = {
-    DRAFT: hasVersion ? [activateAction, deleteAction] : [deleteAction],
+    DRAFT: purpose.mostRecentVersion ? [activateAction, deleteAction] : [deleteAction],
     ACTIVE: [suspendAction, updateDailyCallsAction],
     SUSPENDED: [activateAction, archiveAction, updateDailyCallsAction],
     WAITING_FOR_APPROVAL: [
       purpose.versions.length > 1 ? deleteDailyCallsUpdateAction : deleteAction,
-      updateDailyCallsAction,
     ],
     ARCHIVED: [],
   }
 
-  const status = hasVersion && purpose.mostRecentVersion ? purpose.mostRecentVersion.state : 'DRAFT'
+  const mostRecentVersionState = purpose.mostRecentVersion
+    ? purpose.mostRecentVersion.state
+    : 'DRAFT'
+  const currentVersionState = purpose.currentVersion ? purpose.currentVersion.state : 'DRAFT'
+  const actions = availableActions[mostRecentVersionState]
+
+  // If the most recent version of the purpose is in waiting for approval...
+  if (mostRecentVersionState === 'WAITING_FOR_APPROVAL') {
```

Comment:
Non amo molto l'if nestato ma non ho una soluzione migliore in questo momento, visto il `return { actions }` alla fine

Reply:
Un'alternativa valida potrebbe essere:

```suggestion
if (
    mostRecentVersionState === 'WAITING_FOR_APPROVAL' &&
    purpose.mostRecentVersion?.id !== purpose.currentVersion?.id
  ) {
    actions.push(...availableActions[currentVersionState])
  }

  if (
    mostRecentVersionState === 'WAITING_FOR_APPROVAL' &&
    purpose.mostRecentVersion?.id === purpose.currentVersion?.id
  ) {
    actions.push(updateDailyCallsAction)
  }
```
Un po' ripetitivo ma evita il nesting. Cosa ne pensi?